### DISCUSSION TITLE
compress files from micropython

### DISCUSSION THREAD

**Discussion by lixas** [micropython/micropython#10720] [General]
Hello
I've build web server on ESP32 microcontroller and would like to add availability to make backups of particular files.
What are my options to compress files from micropython??

So far I'm just able to find decompression functionality, but not compression...

**Comment by dlech** [comment]
https://github.com/micropython/micropython/issues/5590

**Comment by bixb922** [comment]
Seems to be technically challenging, there is very interesting and recent talk about this topic: https://m.youtube.com/watch?v=Msfn0Qn_AgI

**Comment by bixb922** [comment]
Have you looked at this? 
https://rosettacode.org/wiki/LZW_compression#Python  Might work for small files, or if you have enough RAM, the dictionary during compression may well need 10 or more times the size of the file.

If the files are data, probably your best option is to write binary files with the smallest or most compact data type available for each data type.

**Comment by jimmo** [comment]
There is WIP support for adding zlib/gzip compression (to match the existing decompression support), hopefully this will be finished after 1.20 is released.

**Comment by jonnor** [comment]
There is now pretty good support for deflate/zlib style compression, including streaming. See the deflate module, https://docs.micropython.org/en/latest/library/deflate.html#deflate.DeflateIO

If one wants to use ZIP archive files, there is https://github.com/jonnor/micropython-zipfile
